File: angular16-boilerplate/src/app/shared/models/company.ts
export class Company {
    public id: number;
    public name: string;
    public address?: string | null;
    public website?: string | null;
    public email?: string | null;
    public phone?: string | null;
  
    constructor(
      id: number,
      name: string,
      address?: string | null,
      website?: string | null,
      email?: string | null,
      phone?: string | null
    ) {
      this.id = id;
      this.name = name;
      this.address = address;
      this.website = website;
      this.email = email;
      this.phone = phone;
    }
  }

File: webapi/DAL/Models/Company.cs
using System;
using System.Collections.Generic;

namespace DAL.Models
{
    public partial class Company
    {
        public Company()
        {
            Agreements = new HashSet<Agreement>();
        }

        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string? Address { get; set; }
        public string? Website { get; set; }
        public string? Email { get; set; }
        public string? Phone { get; set; }

        public virtual ICollection<Agreement> Agreements { get; set; }
    }
}
